src/main/java/org/apache/commons/io/FileUtils.java

```diff
@@ -309,6 +309,9 @@ public static void cleanDirectory(final File directory) throws IOException {
         if (!causeList.isEmpty()) {
             throw new IOExceptionList(causeList);
         }
+        if (listFiles(directory, null).length > 0) {
```

Comment:
Hi @boris-unckel 

Thank you for your PR.

That is an expensive way to test for emptiness if there are a lot of files. Please rebase on master and call `isEmptyDirectory(File)` in this class.

TY.

Reply:
obsolete - it's removed, see later comment